### PR TITLE
Task/bws 105 permissions as list of chips

### PR DESCRIPTION
**Pull request by junojei**
1280x720
localhost_3000_eam(1280x720) (1)
1920x1080
localhost_3000_eam(1920x1080)
localhost_3000_eam(1920x1080) (1)

### CODE REVIEW

File: frontend/src/components/common/chip/chip.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  chipStyle?: ChipStyle;
+};
+
+const Chip: FC<Props> = ({ value, chipStyle = ChipStyle.FILLED }) => {
+  const fullClassName = getValidClasses(
+    styles.chip,
+    styles[`chipStyle${chipStyle}`],
+  );
+
+  return <button className={fullClassName}>{value}</button>;
```
Comment (what1s1ove) [inline frontend/src/components/common/chip/chip.tsx]:
```suggestion
  return <span className={fullClassName}>{value}</span>;
```
let's leave as span for now

File: frontend/src/components/common/chip/chip.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react';
+import { ChipStyle } from 'common/enums/enums';
+import { getValidClasses } from 'helpers/helpers';
+
+import styles from './styles.module.scss';
+
+type Props = {
+  value: string;
```
Comment (what1s1ove) [inline frontend/src/components/common/chip/chip.tsx]:
```suggestion
```

File: frontend/src/components/common/chip/chip.tsx (hunk 1 above)
Comment (what1s1ove) [inline frontend/src/components/common/chip/chip.tsx]:
```suggestion
  return <button className={fullClassName}>{children}</button>;
```

File: frontend/src/components/common/chip/styles.module.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+.chip {
+  min-height: 40px;
+  margin-right: 10px;
+  padding: 10px;
+  color: var(--color-white);
+  text-align: center;
+  background-color: transparent;
+  border: 2px solid transparent;
+  border-radius: 50px;
+  cursor: pointer;
```
Comment (what1s1ove) [inline frontend/src/components/common/chip/styles.module.scss]:
```suggestion
```

File: frontend/src/components/common/chip/chip.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  chipStyle?: ChipStyle;
```
Comment (what1s1ove) [inline frontend/src/components/common/chip/chip.tsx:8]:
```suggestion
  chipStyle?: ChipStyle;
  chipColor??: ChipColor;
```
according to out design in figma

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — APPROVED** [review]

File: frontend/src/components/common/chip/styles.module.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+.chip {
+  min-height: 40px;
+  margin-right: 10px;
+  padding: 10px;
```
Comment (Yaroslaww-1) [inline frontend/src/components/common/chip/styles.module.scss]: I'd make vertical padding +-5px, and horizontal - 10px. Currently Chip component is too large in comparison to other table elements

**Review by Yaroslaww-1 — CHANGES_REQUESTED** [review]

**Comment by junojei** [comment]
1280x720
default
localhost_3000_eam(1280x720) (3)
possible styles
localhost_3000_eam(1280x720) (4)
localhost_3000_eam(1280x720) (5)
localhost_3000_eam(1280x720) (6)

**Comment by junojei** [comment]
fix icon size
image
localhost_3000_eam(1280x720) (10)

File: frontend/src/common/enums/ui/icon-source.enum.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { IconName } from './icon-name.enum';
+import deleteIcon from 'assets/img/delete-icon.svg';
+import editIcon from 'assets/img/edit-icon.svg';
+import keyIcon from 'assets/img/key-icon.svg';
+import reloadIcon from 'assets/img/reload-icon.svg';
+import successIcon from 'assets/img/success-icon.svg';
+import errorIcon from 'assets/img/error-icon.svg';
+import closeIcon from 'assets/img/close-icon.svg';
+
+const IconSource = {
```
Comment (what1s1ove) [inline frontend/src/common/enums/ui/icon-source.enum.ts]: it isn't a enum. this is a map

move it to the `common/maps/ui/`

and name as 
```suggestion
const iconNameToSrc = {
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/common/enums/ui/icon-source.enum.ts (hunk 6 above)
Comment (what1s1ove) [reply frontend/src/common/enums/ui/icon-source.enum.ts]: read more - https://binary-studio.com/2021/08/04/js-and-everyday-data-structures/

**Review by ania-leichenko — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**